src/libraries/System.IO.Compression/tests/ZipArchive/zip_CreateTests.Unix.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Compression.Tests
+{
+    public partial class zip_CreateTests : ZipFileTestBase
+    {
+        [Fact]
+        public static async Task Verify_Default_FilePermissions_Are_Applied_To_Files()
+        {
+            const string testFileContent = "Lorem ipsum dolor sit amet, consectetur adipiscing elit.";
+            const string nestedFolder = "folder/";
+            const string file = "inmemory_file.txt";
+            string fileName = Guid.NewGuid() + ".zip";
+            string outputFolder = Guid.NewGuid().ToString();
+
+            try
+            {
+                await CreateArchive();
+                await UnpackArchive();
+
+                EnsureFilePermissions(Path.Combine(outputFolder, nestedFolder, file), "644");
```

Comment:
and check folder is 755?

Reply:
Ideally, yes. But with the current unzipping implementation, we can't do that: the folders are just created using [`Directory.CreateDirectory`](https://cs.github.com/dotnet/runtime/blob/5877e8b713742b6d80bd1aa9819094be029e3e1f/src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs#L110) . When `unzip` is used, the permissions are fully respected of course.
IMO changing the unzipping logic fits this issue better:
 - https://github.com/dotnet/runtime/issues/71140

Reply:
We can reduce what these tests are doing to: creating a `ZipArchiveEntry`, and verifying `ExternalAttributes` has the expected value.
That will also work for directories.

Reply:
The existing `UnixExtractSetsFilePermissionsFromExternalAttributes` verifies `ExternalAttributes` gets stored and restored.

Reply:
@tmds I made a very simple test instead of the one I had previously. Is this what you had in mind?